luigi/worker.py

```diff
@@ -971,10 +971,18 @@ def _handle_next_task(self):
                     self.add(t)
                 new_deps = [t.task_id for t in new_req]
 
+            try:
+                expl_json = json.dumps(expl)
+            except UnicodeDecodeError:
+                logger.info('Unicode decode error occur during %s task metadata serialisation, '
+                            'falling back to latin1 encoding.', task_id)
+
+                expl_json = json.dumps(expl, encoding='latin1')
```

Comment:
Hm.. I'm not sure that proposed solution is good enough to go with due to number of reasons.

* First, according to [RFC 7159](https://tools.ietf.org/html/rfc7159#page-9), JSON text SHALL be encoded in UTF-8, UTF-16, or UTF-32. Encoding it in `latin1` may break consumers one day.

* Second, what do you expect to see if you pass Python 2 `str` (which is bytes) to `json.dumps()`? Obviously, bytes may contain unserializable characters. Instead, I think it would make more sense to craft a `unicode` (in Python 2) object instead. Can we figure out why exactly `expl` is `str` in py2?